packages/core/src/model.ts

```diff
@@ -6,24 +6,82 @@ import type {
   ExtractEvent,
   EventObject
 } from './types';
+import { mapValues } from './utils';
 
-export interface ContextModel<TContext, TEvent extends EventObject> {
+export interface ContextModel<
+  TContext,
+  TEvent extends EventObject,
+  TEM extends EventCreatorMap<TEvent> = EventCreatorMap<TEvent>
+> {
   initialContext: TContext;
   assign: <TEventType extends TEvent['type'] = TEvent['type']>(
     assigner:
       | Assigner<TContext, ExtractEvent<TEvent, TEventType>>
       | PropertyAssigner<TContext, ExtractEvent<TEvent, TEventType>>,
     eventType?: TEventType
   ) => AssignAction<TContext, ExtractEvent<TEvent, TEventType>>;
+  events: FullEventCreatorMap<TEM>;
+  types: {
+    context: TContext;
+    events: TEvent;
```

Comment:
Is there prior art on this? I would like to avoid the confusion... maybe have these under `types`?

Thinking about it, these might not be necessary. We can make a helper, e.g.,:

```ts
type EventsOf<TModel> = TModel extends Model<any, infer Event> ? Event : never;
```

Reply:
Omitted `types` - using `ModelContextFrom<Model>` and `ModelEventsFrom<Model>` instead